sys/shell/cmds/rtc.c

```diff
@@ -39,10 +40,24 @@ static int dow(int year, int month, int day)
 {
     /* calculate the day of week using Tøndering's algorithm */
     static int t[] = {0, 3, 2, 5, 0, 3, 5, 1, 4, 6, 2, 4};
```

Comment:
Unrelated optimization proposal: this could be
```suggestion
    static const int8_t t[] = {0, 3, 2, 5, 0, 3, 5, 1, 4, 6, 2, 4};
```

Reply:
(or even `uint8_t` but that's less about optimization and more type correctness)

Reply:
Why is this even needed?
`rtc_tm_normalize()` will already do this - no need to implement this logic in a shell command.

Reply:
Hm, indeed rtc_utils.c's `_wday` is a duplicate of this function, with different properties (it already has the const uint8_t type; it doesn't perform overflow checks but then again is only called with pre-sanitized input).

Honestly, if it was for me, `struct tm` would have no place in an embedded OS at all -- I won't impede work on it, and am happy to handle security issues arising from it (even biting down any snark), but addressing this duplication will require judgement calls on "Should the shell RTC command do the full normalization `rtc_tm_normalize` performs?" and "Should the shell RTC command pull in `RTC_NORMALIZE_COMPAT=1`?", and my answer to both would be [mu](https://en.wikipedia.org/wiki/Mu_(negative)#Non-dualistic_meaning).